mysql_protocol.cc

```diff
@@ -187,14 +193,51 @@ bool Protocol::Authenticate() {
     // there should atleast be a \0 byte as password
     return false;
   }
-  const char *passwd = user + user_len + 1;
-  int passwd_len = Unpack(&passwd);  // length encoded
+  const char *authdata = user + user_len + 1;
+  int authdata_len = Unpack(&authdata);  // length encoded
```

Comment:
Should there be some verification that authdata_len is at least SHA_DIGEST_LENGTH or bigger?
I understand that there wasn't such check before, but it would be great to add it, or completely remove this line since authdata_len is not used anywhere now.